### PR TITLE
Support 'pending' block in RPC

### PR DESCRIPTION
**Pull request by arkpar**
Partially takes care of #926

### CODE REVIEW

File: miner/src/miner.rs [hunk 1]
```diff
@@ -175,6 +176,51 @@ impl MinerService for Miner {
...
+		// give the sender max balance
+		state.sub_balance(&sender, &balance);
+		state.add_balance(&sender, &U256::max_value());
+		let options = TransactOptions { tracing: false, check_nonce: false };
+		Executive::new(&mut state, &env_info, chain.engine()).transact(t, options)
+	}
+
+	fn balance(&self, address: &Address) -> U256 {
+		let sealing_work = self.sealing_work.lock().unwrap();
+		sealing_work.peek_last_ref().map_or(x!(0), |b| b.block().fields().state.balance(address))
```
Comment (tomusdrw) [inline miner/src/miner.rs]: Maybe you could use `self.map_sealing_work` this would actually enable and prepare sealing?

Comment (arkpar) [inline miner/src/miner.rs]: There is no need to actually start sealing here

Comment (gavofyork) [inline miner/src/miner.rs]: what about when there's no work (`peer_last_ref()` returns `None`) and it is querying an account with a non-zero balance?

Comment (arkpar) [inline miner/src/miner.rs]: Now miner will forward calls to client if not sealing. Checking sealing status in RPC module would be a subject to race condition. This whole module needs to be refactored later. Miner module should keep a reference to `BlockChainClient`

**Comment by gavofyork** [comment]
looks better now - are there any tests?

**Comment by arkpar** [comment]
Tests would require substantial refactoring of `TestMinerService` and `TestBlockChainClient`. Tested manually for now

**Pull request merged**